---
#date: "`r Sys.Date()`"
output:
  pagedown::html_paged:
    # offline
    css: ["css/one-column-paged.css"]
    # "https://use.fontawesome.com/releases/v5.7.2/css/all.css"]
    number_sections: false
title: "Emma Dann"
links-to-footnotes: false
knit: pagedown::chrome_print
---

```{css, echo=FALSE}
a {
  color: #710C0C
}
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
library(tidyverse)
library(knitr)
library(lubridate)
library(fontawesome)
```


<center>

`r fa("map-marker-alt", fill = '#710C0C')`  Wellcome Sanger Institute, Cambridge (UK) \ | \  `r fa("envelope", fill = '#710C0C')`  emma.dann\@sanger.ac.uk  \
`r fa("globe", fill = '#710C0C')` [emdann.github.io](https://emdann.github.io) \ | \ 
`r fa("github", fill = '#710C0C')`   [emdann](https://github.com/emdann) \ | \ 
`r fa("twitter", fill = '#710C0C')`   [\@emmamarydann](https://twitter.com/emmamarydann)  

</center>

<!-- <div class="grid-container">  -->
<!-- top row -->
<!-- ```{block type="grid-item", echo=TRUE} -->
<!-- ## Personal summary {.summary} -->
<!-- I’m a PhD student in computational biology, working on statistical methods for complex single-cell genomics datasets. I'm passionate about open science and equality in and out of science. -->
<!-- ``` -->

<!-- ```{block type='grid-item', echo=TRUE} -->
<!-- ## Contact information {.box} -->
<!-- <div class="info-box"> -->

<!-- | Center | Left | -->
<!-- |:------:|:-----| -->
<!-- `r fa("map-marker-alt", fill = "steelblue")`  Wellcome Sanger Institute, Cambridge (UK) -->
<!-- `r fa("envelope", fill = "steelblue")`   emma.dann\@sanger.ac.uk  -->
<!-- `r fa("globe", fill = "steelblue")`   [emdann.github.io](https://emdann.github.io)  -->
<!-- `r fa("github", fill = "steelblue")`   [emdann](https://github.com/emdann)  -->
<!-- `r fa("twitter", fill = "steelblue")`   [\@emmamarydann](https://twitter.com/emmamarydann)  -->
 
<!-- </div> -->
<!-- ``` -->

<!-- </div> -->

## Education {.education .nohead .set-width}

**PhD in Biological Sciences (Computational Biology)** -- University of Cambridge, UK (2019-present)  
Supervisors: Dr. Sarah A. Teichmann, Dr. John C. Marioni  
<!-- Cambridge University advisor: Dr. Neil D. Lawrence -->

**M.Sc. in Cancer, Stem Cells and Developmental Biology (Bioinformatics profile)** -- Utrecht University, NL (2017-2019)  
Final grade: _cum laude_ (Honours, Avg. grade: 9.10/10) 

**B.Sc. in Biomolecular Sciences and Technology** -- University of Trento, IT  (2014-2017)  
Final grade: 110 _cum laude_/110 
<!-- ### B.Sc. in Biomolecular Sciences and Technology - University of Trento -->

<!-- Final grade: 110 _cum laude_/110  -->

<!-- Trento, IT -->

<!-- 2014 -->

## Professional experience
**Graduate researcher -- Wellcome Sanger Institute, Cambridge, UK** (2019 -- present)  
[Teichmann lab](http://www.teichlab.org/) / [Marioni group](https://www.ebi.ac.uk/research/marioni) -- Computational methods for comparative single-cell analysis of developmental atlases
<!-- - Differential cell abundance analysis -->
<!-- - Analysis of human developmental atlases -->
<!-- - Methods for joint scATAC-seq and scRNA-seq data -->

**Research intern -- European Molecular Biology Laboratory, Heidelberg, DE** (2018 -- 2019)  
[Huber group](https://www.huber.embl.de/) -- Integrative analysis of multi-omic profiles of perturbations for functional characterization of drug treatments and target discovery.

**Research intern -- KNAW Hubrecht Institute, Utrecht, NL** (2017 -- 2018)  
[Van Oudenaarden lab](https://www.hubrecht.eu/research-groups/van-oudenaarden-group/) -- Biophysical modelling of whole-genome amplification via random priming, to predict and enhance coverage of single-cell bisulfite sequencing experiments.

**Research intern -- Center for Integrative Biology, University of Trento, IT** (2017)  
[Demichelis lab](http://demichelislab.eu/) -- Computational analysis of whole-methylome profiles of tumor samples from a large cohort of prostate cancer patients, for epigenetic biomarker discovery. 

**Research assistant -- Center for Integrative Biology, University of Trento, IT** (2016)  
Lab equipment maintenance and general support for benchwork.

## Software

**miloR** | <i class="fab fa-github"></i> [MarioniLab/miloR](https://github.com/MarioniLab/miloR) | primary developer | R/Bioconductor package for the identification of differential cell abundance in KNN graphs for single-cell data

**milopy** | <i class="fab fa-github"></i> [emdann/milopy](https://github.com/emdann/milopy) | primary developer | python package for the identification of differential cell abundance in KNN graphs for single-cell data

**oor_benchmark** |  <i class="fab fa-github"></i> [MarioniLab/oor_benchmark](https://github.com/emdann/oor_benchmark) | primary developer | python framework to benchmark workflows for detection of out-of-reference cells in comparative single-cell analysis 

**cell2location** | <i class="fab fa-github"></i> [BayraktarLab/cell2location](https://github.com/BayraktarLab/cell2location) | contributor | python package for mapping of tissue cell types via integration of single cell RNA-seq and spatial transcriptomics data

**SingleCellOpenProblems** | <i class="fab fa-github"></i> [singlecellopenproblems/SingleCellOpenProblems](https://github.com/singlecellopenproblems/SingleCellOpenProblems) | contributor | A python framework to formalize and benchmark open problems in single-cell genomics data analysis.

## Publications

Suo C., Polanski K., **Dann E.**, ... ,  Tuong Z.K., Clatworthy M., Teichmann S.A. (2022) Single cell antigen receptor analysis reveals lymphocyte developmental origins, _biorXiv_ | [Full text](https://www.biorxiv.org/content/10.1101/2022.11.18.517068v1) | 
[python package](https://github.com/zktuong/dandelion)

**Dann E.**, Teichmann S.A., Marioni J.C. (2022) Precise identification of cell states altered in disease with healthy single-cell references, _biorXiv_ | [Full text](https://www.biorxiv.org/content/10.1101/2022.11.10.515939v1) | 
[code](https://github.com/MarioniLab/oor_design_reproducibility) | 
[python package](https://github.com/MarioniLab/oor_benchmark)

Lalchand V.\*, Ravuri A.\*, __Dann E.__\*, ... , Teichmann S.A., Lawrence N.D., Modelling Technical and Biological Effects in scRNA-seq data with Scalable GPLVMs. _arXiv_ | [Full text](https://arxiv.org/pdf/2209.06716.pdf)   
<sup>\*Equal Contribution</sup>  

Botting R.A., Goh I., ... , **Dann E.**, ... , Teichmann S.A. , Haniffa M., Multi-organ functions of yolk sac during human early development. _biorXiv_ |
[Full text](https://www.biorxiv.org/content/10.1101/2022.08.03.502475v1.full.pdf) 

Suo C.\*, __Dann E.__\*, ... , Clatworthy M., Haniffa M., Teichmann S.A. (2022), Mapping the developing human immune system across organs. _Science_ |
[Full text](https://www.science.org/doi/10.1126/science.abo0510#con2) | [code](https://github.com/Teichlab/Pan_fetal_immune) | [data](https://developmental.cellatlas.io/fetal-immune)  
<sup>\*Equal Contribution</sup>  

Kleshchevnikov V., Shmatko A., **Dann E.**, ..., Stegle O.,  Bayraktar O.A. (2022) Cell2location maps fine-grained cell types in spatial transcriptomics. _Nature Biotechnology_ |
[Full text](https://doi.org/10.1038/s41587-021-01139-4) |
[python package](https://cell2location.readthedocs.io/)

He P., Lim K., Sun D., ... , **Dann E.**, ... , Meyer K.B., Rawlins E.L. (2022) A human fetal lung cell atlas uncovers proximal-distal gradients of differentiation and key regulators of epithelial fates, _bioRxiv_ |
[Full text](https://doi.org/10.1101/2022.01.11.474933)

**Dann E.**, Henderson N.C., Teichmann S.A., Morgan M.D., Marioni J.C. (2021) Differential abundance testing on single-cell data using k-nearest neighbor graphs, _Nature Biotechnology_ |
[Full text](https://www.nature.com/articles/s41587-021-01033-z) | 
[R package](https://github.com/MarioniLab/miloR) | 
[python package](https://github.com/emdann/milopy)

Jardine L., Webb S., ... , **Dann E.**, ... , Göttgens B., Roberts I., Teichmann S. and Haniffa M. (2021) Blood and immune development in human fetal bone marrow and Down syndrome, _Nature_ |
[Full text](https://doi.org/10.1038/s41586-021-03929-x)

Elmentaite R., Kumasaka N., Roberts K., Fleming A., **Dann E.**, ... , James K.R., Teichmann S.A. (2021), Cells of the human intestinal tract mapped across space and time, _Nature_ |
[Full text](https://www.nature.com/articles/s41586-021-03852-1)

Stephenson E., Reynolds G., Botting R.A., Calero-Nieto F.J., Morgan M.D., Tuong Z.K., Bach K., Sungnak W., ... , **Dann E.**, ... , Duncan C.J.A, Smith K., Teichmann S.A., Clatworthy M.R., Marioni J.C., Gottgens B., Haniffa M. (2020) Single-cell multi-omics analysis of the immune response in COVID-19, _Nature Medicine_ |
[Full text](https://www.nature.com/articles/s41591-021-01329-2)

<!-- Kleshchevnikov V., Shmatko A., **Dann E.**, ..., Stegle O.,  Bayraktar O.A. (2020) Comprehensive mapping of tissue cell architecture via integrated single cell and spatial transcriptomics, bioRxiv, [doi: https://doi.org/10.1101/2020.11.15.378125](https://doi.org/10.1101/2020.11.15.378125) -->

Beltran H., Romanel A., ... , **Dann E.**, ... , Benelli M., and Demichelis F. (2020) Circulating Tumor DNA Profile Recognizes Transformation to Castration-Resistant Neuroendocrine Prostate Cancer, _Journal of Clinical Investigation_ |
[Full text](https://www.jci.org/articles/view/131041)

<!-- \newpage -->
## Conference presentations

Single Cell Genomics | Utrecht (NL) | October 2022    
_Precise disease-state identification with healthy single-cell references_ (poster presentation)

UK Conference of Bioinformatics and Computational Biology | virtual | September 2022  
_Mapping the developing immune system across organs_ (invited talk)

Single Cell Biology | Wellcome Genome Campus, Hinxton (UK) | June 2022  
_Mapping the developing immune system across organs_ (selected talk)

The Biology of Genomes | Cold Spring Harbor Laboratory (US) | May 2022  
_Mapping the developing immune system across organs_ (selected talk)

Maxwell Society Annual Conference | King's College London (UK) | Apr 2022  
_Best practices and open challenges in single-cell genomics data analysis_ (invited talk)

Chan Zuckerberg Initiative Single Cell Biology Meeting | virtual | Oct 2021  
_Milo: differential abundance testing on single-cell data using KNN graphs_ (selected talk)

Next Generation Genomics Meeting | virtual | Sept 2021  
_Differential abundance testing on single-cell data using KNN graphs_ (selected talk) ([video](https://underline.io/lecture/36963-milo-differential-abundance-testing-on-single-cell-data-using-knn-graphs))

Human Cell Atlas General Meeting | virtual | Jun 2021  
_Mapping human immune system development across fetal organs_ (poster presentation)

Cambridge Center for Physical Biology Single Cell Symposium | virtual | Mar 2021  
_Milo: differential abundance testing on single-cell data using KNN graphs_ (invited talk)

Chan Zuckerberg Initiative Seed Networks Computational biology meeting | virtual | Mar 2021  
_Milo: differential abundance testing on single-cell data using KNN graphs_ (short talk)

<!-- <br class="pageBreak"> -->

<!-- ## Courses -->

<!-- **Gaussian Process and Uncertainty Quantification Summer School** (2020) -->

<!-- **Accelerate Science Winter School** (2021) -->

## Other projects

[Multimodal single-cell best practices](www.sc-best-practices.org) (2022) | Chapter contributor (compositional analysis)

[Systems biology: From large datasets to biological insight - EMBL-EBI Training](https://www.ebi.ac.uk/training/materials/systems-biology-large-datasets-biological-insight-handbook-2022/) (2022) |
Instructor for single-cell multi-omics module

[Advanced topics in Single Cell Genomics - SIB course](https://www.sib.swiss/training/course/20220426_ADVSC) (2022) |
Trainer for multi-omics module

[Advanced topics in Single Cell Genomics - Joint SciLifeLab-SIB Summer School](https://nbisweden.github.io/single-cell_sib_scilifelab_2021/) (2021) |
Trainer for multi-omics project

[Open Problems in Single-Cell Analysis - Spring Jamboree](https://openproblems.bio/jamboree/) (2021) |
Co-organizer and hacker.

[EBI-Sanger-Cambridge PhD Symposium](https://escamps.org/escamps20/) (2020) |
Member of the organizing committee.

[OWL - Open Wet Lab](http://www.openwetlab.it/home-en/) (2015 -- 2018) |
<!-- Board member and director of scientific outreach blog -->
OWL is the first biohacking organization in Italy. I served as board member and directed the publication of OWL's blog and scientific outreach articles for local newspapers. 

[Model European Parliament](https://mepeurope.eu) (2012 -- 2015) |
International simulation of the working of the European Parliament for high school students. I was delegate, committee president and president of the general assembly at many local and international MEP sessions. 

<!-- ## Academic service -->

<!-- **Peer review** -->
<!-- _Nature Biotechnology_ -->
<!-- _Bioinformatics_ -->

## Fellowships and awards

4-year Wellcome Sanger Institute PhD studentship (2019-2023)

U/Select honours programme and travel grant - Utrecht University  (2018)  
<!-- Honours programme for selected students of the Graduate School of Life Sciences, incl. travel grant to perform M.Sc. research project abroad -->

Merit Award - University of Trento (2017)  
<!-- Cash prize for excellent graduates established to reward their skills and their commitment to study -->

<!-- ## Academic referees -->

<!-- - Dr Sarah A. Teichmann FMedSci FRS - Head of Cellular Genetics programme st9<at>sanger.ac.uk -->
<!-- - Dr John C. Marioni - marioni<at>ebi.ac.uk -->
<!-- - Dr Wolfgang Huber - wolfgang.huber<at>embl.de -->
