# Publications

<!-- <br>
<h3>Preprints</h3>
<br> -->

See my [Google Scholar](https://scholar.google.com/citations?hl=en&user=J2NQ29wAAAAJ&view_op=list_works) for the most up-to-date list.

<br>
<h3>Key publications</h3>
<br>

__Dann E.__\*, Teeple E.\*, Elmentaite R., Meyer K., Gaglia G., Nestle F., Savova V., Rinaldis E., Teichmann S.A. (2024) Estimating the impact of single-cell RNA sequencing of
human tissues on drug target validation., _medRxiv_ | [Full text](https://www.medrxiv.org/content/10.1101/2024.04.04.24305313v2.full.pdf) | 
[code](https://github.com/emdann/sc_target_evidence)  
<sup>\*Equal Contribution</sup>

**Dann E.**, Cujba A.M., Oliver A.J., Meyer K., Teichmann S.A., Marioni J.C. (2022) Precise identification of cell states altered in disease with healthy single-cell references, _Nature Genetics_ | [Full text](https://www.nature.com/articles/s41588-023-01523-7) | 
[code](https://github.com/MarioniLab/oor_design_reproducibility) | 
[python package](https://github.com/MarioniLab/oor_benchmark)

Suo C.\*, __Dann E.__\*, ... , Haniffa M., Teichmann S.A. (2022) Mapping the developing human immune system across organs. _Science_ |
[Full text](https://www.science.org/doi/10.1126/science.abo0510#con2) | [code](https://github.com/Teichlab/Pan_fetal_immune) | [data](https://developmental.cellatlas.io/fetal-immune)  
<sup>\*Equal Contribution</sup>  

**Dann E.**, Henderson N.C., Teichmann S.A., Morgan M.D., Marioni J.C. (2021) Differential abundance testing on single-cell data using k-nearest neighbor graphs, _Nature Biotechnology_ |
[Full text](https://www.nature.com/articles/s41587-021-01033-z) | 
[R package](https://github.com/MarioniLab/miloR) | 
[python package](https://github.com/emdann/milopy)

<br>
<h3>Other publications</h3>
<br>

Oliver A.J., Ni H., ... , **Dann E.**, ..., Elmentaite R., Teichmann S.A. (2024) Single Cell Integration Characterises Gastric Cell Metaplasia in Inflammatory Intestinal Diseases. _biorXiv (accepted at Nature)_ | [Full text](https://www.biorxiv.org/content/10.1101/2024.10.11.614415v1.abstract)

Heumos L., Ji Y., ... , Dann E., ... , Theis F.J. Pertpy: an end-to-end framework for perturbation analysis. _biorXiv_ | [Full text](https://www.biorxiv.org/content/10.1101/2024.08.04.606516v1.full) | [python package](https://github.com/scverse/pertpy)

Missarova A., **Dann E.**, ... , J.C. Marioni (2023) Leveraging neighborhood representations of single-cell data to achieve sensitive DE testing with miloDE. _Genome Biology_ | [Full text](https://link.springer.com/article/10.1186/s13059-024-03334-3) | [R package](https://github.com/MarioniLab/miloDE)

Sumanaweera D., Suo C., Muraro D., **Dann E.**, ... , Teichmann S.A. (2023) Gene-level alignment of single cell trajectories. _Nature Methods_ | [Full text](https://www.nature.com/articles/s41592-024-02378-4) | [python package](https://github.com/Teichlab/Genes2Genes)

Luecken M., Gigante S., Burkhardt D., Cannoodt R., ... , **Dann E.**, ... , Saeys Y., Theis F., Krishnaswamy S. (2024) Defining and benchmarking open problems in single-cell analysis. _Research Square_ | [Full text](https://pmc.ncbi.nlm.nih.gov/articles/PMC11030530/)

Yayon N., Kedlian V., Boehme L., ... , **Dann E**, ... , Uhlmann V., Notarangelo L.D., Germain R.N., Radtke A.J., Marioni J.C., Taghon T., Teichmann S.A. (2023) A spatial human thymus cell atlas mapped to a continuous tissue axis. _bioRxiv_ | [Full text](https://www.biorxiv.org/content/10.1101/2023.10.25.562925v1)

Schuster V., **Dann E.**, Krogh A., Teichmann S.A. (2023) multiDGD: A versatile deep generative model for multi-omics data. _biorXiv (accepted at Nature Communications)_ | [Full text](https://www.biorxiv.org/content/10.1101/2023.08.23.554420v1.full) | [python package](https://github.com/Center-for-Health-Data-Science/multiDGD)

Suo C., Polanski K., **Dann E.**, ... ,  Tuong Z.K., Clatworthy M., Teichmann S.A. (2022) Dandelion uses the single-cell adaptive immune receptor repertoire to explore lymphocyte developmental origins, _Nature Biotechnology_  | [Full text](https://www.nature.com/articles/s41587-023-01734-7) | 
[python package](https://github.com/zktuong/dandelion)

Barnes J. L., He P., ... , **Dann E.**, ... , Teichmann S.A., Meyer K. B., Nikolic M. Z.  (2022) Early human lung immune cell development and its role in epithelial cell fate. _Science Immunology_ | [Full text](https://www.science.org/doi/abs/10.1126/sciimmunol.adf9988)

Heumos L., ... **Single-cell Best Practices Consortium**, ... , Theis F.J. (2023) Best practices for single-cell analysis across modalities. _Nature Review Genetics_ | [full text](https://www.nature.com/articles/s41576-023-00586-w) 

He P., Lim K., Sun D., ... , **Dann E.**, ... , Meyer K.B., Rawlins E.L. (2022) A human fetal lung cell atlas uncovers proximal-distal gradients of differentiation and key regulators of epithelial fates, _Cell_ |
[Full text](https://www.cell.com/cell/fulltext/S0092-8674(22)01415-5?_returnURL=https%3A%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS0092867422014155%3Fshowall%3Dtrue)

Lalchand V.\*, Ravuri A.\*, __Dann E.__\*, ... , Teichmann S.A., Lawrence N.D. (2022) Modelling Technical and Biological Effects in scRNA-seq data with Scalable GPLVMs. _Proceedings of the 17th Machine Learning in Computational Biology meeting_ | [Full text](https://proceedings.mlr.press/v200/lalchand22a)   
<sup>\*Equal Contribution</sup>  

Botting R.A., Goh I., ... , **Dann E.**, ... , Teichmann S.A. , Haniffa M. (2022) Multi-organ functions of yolk sac during human early development. _biorXiv_ |
[Full text](https://www.biorxiv.org/content/10.1101/2022.08.03.502475v1.full.pdf) 

Kleshchevnikov V., Shmatko A., **Dann E.**, ..., Stegle O.,  Bayraktar O.A. (2022) Cell2location maps fine-grained cell types in spatial transcriptomics. _Nature Biotechnology_ |
[Full text](https://doi.org/10.1038/s41587-021-01139-4) |
[python package](https://cell2location.readthedocs.io/)

Jardine L., Webb S., ... , **Dann E.**, ... , GÃ¶ttgens B., Roberts I., Teichmann S. and Haniffa M. (2021) Blood and immune development in human fetal bone marrow and Down syndrome, _Nature_ |
[Full text](https://doi.org/10.1038/s41586-021-03929-x)

Elmentaite R., Kumasaka N., Roberts K., Fleming A., **Dann E.**, ... , James K.R., Teichmann S.A. (2021), Cells of the human intestinal tract mapped across space and time, _Nature_ |
[Full text](https://www.nature.com/articles/s41586-021-03852-1)

Stephenson E., Reynolds G., Botting R.A., Calero-Nieto F.J., Morgan M.D., Tuong Z.K., Bach K., Sungnak W., ... , **Dann E.**, ... , Duncan C.J.A, Smith K., Teichmann S.A., Clatworthy M.R., Marioni J.C., Gottgens B., Haniffa M. (2020) Single-cell multi-omics analysis of the immune response in COVID-19, _Nature Medicine_ |
[Full text](https://www.nature.com/articles/s41591-021-01329-2)

Beltran H., Romanel A., ... , **Dann E.**, ... , Benelli M., and Demichelis F. (2020) Circulating Tumor DNA Profile Recognizes Transformation to Castration-Resistant Neuroendocrine Prostate Cancer, _Journal of Clinical Investigation_ |
[Full text](https://www.jci.org/articles/view/131041)

<!-- <br>
<h3>Review Articles</h3>
<br>

<br>
<h3>Refereed Workshop Papers</h3>
<br> -->


----
