<!doctype html>
<html lang="en">
  <head>
    <!-- Required meta tags -->
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

    <!-- Bootstrap CSS -->
    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.5.0/css/bootstrap.min.css">

    <link href="https://fonts.googleapis.com/css2?family=Roboto+Mono&display=swap" rel="stylesheet">
    <style type="text/css">body {font-family: 'Roboto Mono', serif;}</style>

    <style>
      @media screen and (min-width: 768px) {
        .container {
          width: 768px;
        }
      }

      body {
        background-color: #f5e9dd;
      }
    </style>

    <title>Publications ðŸ“œ</title>

    
    
    
  </head>
  <body>

    
    <div class="container d-none d-lg-block d-xl-block">
      <div class="row mt-5">
        <div class="col-7 mt-5">
          <h1>Publications ðŸ“œ</h1>
          <hr style="max-width: 200px; margin-left:0;" style="width:60%;"/>
          <div class="my-3" id="icon-list">
            <ul class="list-inline">
                          </ul>
          </div>
        </div>

        <div class="col mr-4 mt-5">
          <img src="" style="height:15rem" class="rounded float-right">
        </div>
      </div>
      <div class="row mb-5">
        <div class="col">
          <div id="key-publications" class="section level4">
          <h4>Key publications</h4>
          <p><strong>Dann E.</strong>, Teichmann S.A., Marioni J.C. (2022) Precise identification of cell states altered in disease with healthy single-cell references, <em>biorXiv (accepted at Nature Genetics)</em> | <a href="https://www.biorxiv.org/content/10.1101/2022.11.10.515939v1">Full text</a> | <a href="https://github.com/MarioniLab/oor_design_reproducibility">code</a> | <a href="https://github.com/MarioniLab/oor_benchmark">python package</a></p>
          <p>Suo C.*, <strong>Dann E.</strong>*, â€¦ , Haniffa M., Teichmann S.A. (2022) Mapping the developing human immune system across organs. <em>Science</em> | <a href="https://www.science.org/doi/10.1126/science.abo0510#con2">Full text</a> | <a href="https://github.com/Teichlab/Pan_fetal_immune">code</a> | <a href="https://developmental.cellatlas.io/fetal-immune">data</a><br />
          <sup>*Equal Contribution</sup></p>
          <p><strong>Dann E.</strong>, Henderson N.C., Teichmann S.A., Morgan M.D., Marioni J.C. (2021) Differential abundance testing on single-cell data using k-nearest neighbor graphs, <em>Nature Biotechnology</em> | <a href="https://www.nature.com/articles/s41587-021-01033-z">Full text</a> | <a href="https://github.com/MarioniLab/miloR">R package</a> | <a href="https://github.com/emdann/milopy">python package</a></p>
          </div>
          <div id="other" class="section level4">
          <h4>Other</h4>
          <p>Schuster V., <strong>Dann E.</strong>, Krogh A., Teichmann S.A. (2023) multiDGD: A versatile deep generative model for multi-omics data. <em>biorXiv</em> | <a href="https://www.biorxiv.org/content/10.1101/2023.08.23.554420v1.full">Full text</a> | <a href="https://github.com/Center-for-Health-Data-Science/multiDGD">python package</a></p>
          <p>Suo C., Polanski K., <strong>Dann E.</strong>, â€¦ , Tuong Z.K., Clatworthy M., Teichmann S.A. (2022) Dandelion uses the single-cell adaptive immune receptor repertoire to explore lymphocyte developmental origins, <em>Nature Biotechnology</em> | <a href="https://www.nature.com/articles/s41587-023-01734-7">Full text</a> | <a href="https://github.com/zktuong/dandelion">python package</a></p>
          <p>Heumos L., â€¦ <strong>Single-cell Best Practices Consortium</strong>, â€¦ , Theis F.J. (2023) Best practices for single-cell analysis across modalities. <em>Nature Review Genetics</em> | <a href="https://www.nature.com/articles/s41576-023-00586-w">full text</a></p>
          <p>Missarova A., <strong>Dann E.</strong>, â€¦ , J.C. Marioni (2023) Sensitive cluster-free differential expression testing. <em>biorXiv</em> | <a href="https://www.biorxiv.org/content/10.1101/2023.03.08.531744v1">Full text</a> | <a href="https://github.com/MarioniLab/miloDE">R package</a></p>
          <p>Sumanaweera D., Suo C., Muraro D., <strong>Dann E.</strong>, â€¦ , Teichmann S.A. (2023) Gene-level alignment of single cell trajectories informs the progression of in vitro T cell differentiation. <em>biorXiv</em> | <a href="https://www.biorxiv.org/content/10.1101/2023.03.08.531713v1">Full text</a> | <a href="https://github.com/Teichlab/Genes2Genes">python package</a></p>
          <p>Barnes J. L., He P., â€¦ , <strong>Dann E.</strong>, â€¦ , Teichmann S.A., Meyer K. B., Nikolic M. Z. (2022) Early human lung immune cell development and its role in epithelial cell fate. <em>biorXiv</em> | <a href="https://www.biorxiv.org/content/biorxiv/early/2022/12/15/2022.12.13.519713.full.pdf">Full text</a></p>
          <p>He P., Lim K., Sun D., â€¦ , <strong>Dann E.</strong>, â€¦ , Meyer K.B., Rawlins E.L. (2022) A human fetal lung cell atlas uncovers proximal-distal gradients of differentiation and key regulators of epithelial fates, <em>Cell</em> | <a href="https://www.cell.com/cell/fulltext/S0092-8674(22)01415-5?_returnURL=https%3A%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS0092867422014155%3Fshowall%3Dtrue">Full text</a></p>
          <p>Lalchand V.*, Ravuri A.*, <strong>Dann E.</strong>*, â€¦ , Teichmann S.A., Lawrence N.D. (2022) Modelling Technical and Biological Effects in scRNA-seq data with Scalable GPLVMs. <em>Proceedings of the 17th Machine Learning in Computational Biology meeting</em> | <a href="https://proceedings.mlr.press/v200/lalchand22a">Full text</a><br />
          <sup>*Equal Contribution</sup></p>
          <p>Botting R.A., Goh I., â€¦ , <strong>Dann E.</strong>, â€¦ , Teichmann S.A. , Haniffa M. (2022) Multi-organ functions of yolk sac during human early development. <em>biorXiv</em> | <a href="https://www.biorxiv.org/content/10.1101/2022.08.03.502475v1.full.pdf">Full text</a></p>
          <p>Kleshchevnikov V., Shmatko A., <strong>Dann E.</strong>, â€¦, Stegle O., Bayraktar O.A. (2022) Cell2location maps fine-grained cell types in spatial transcriptomics. <em>Nature Biotechnology</em> | <a href="https://doi.org/10.1038/s41587-021-01139-4">Full text</a> | <a href="https://cell2location.readthedocs.io/">python package</a></p>
          <p>Jardine L., Webb S., â€¦ , <strong>Dann E.</strong>, â€¦ , GÃ¶ttgens B., Roberts I., Teichmann S. and Haniffa M. (2021) Blood and immune development in human fetal bone marrow and Down syndrome, <em>Nature</em> | <a href="https://doi.org/10.1038/s41586-021-03929-x">Full text</a></p>
          <p>Elmentaite R., Kumasaka N., Roberts K., Fleming A., <strong>Dann E.</strong>, â€¦ , James K.R., Teichmann S.A. (2021), Cells of the human intestinal tract mapped across space and time, <em>Nature</em> | <a href="https://www.nature.com/articles/s41586-021-03852-1">Full text</a></p>
          <p>Stephenson E., Reynolds G., Botting R.A., Calero-Nieto F.J., Morgan M.D., Tuong Z.K., Bach K., Sungnak W., â€¦ , <strong>Dann E.</strong>, â€¦ , Duncan C.J.A, Smith K., Teichmann S.A., Clatworthy M.R., Marioni J.C., Gottgens B., Haniffa M. (2020) Single-cell multi-omics analysis of the immune response in COVID-19, <em>Nature Medicine</em> | <a href="https://www.nature.com/articles/s41591-021-01329-2">Full text</a></p>
          <!-- Kleshchevnikov V., Shmatko A., **Dann E.**, ..., Stegle O.,  Bayraktar O.A. (2020) Comprehensive mapping of tissue cell architecture via integrated single cell and spatial transcriptomics, bioRxiv, [doi: https://doi.org/10.1101/2020.11.15.378125](https://doi.org/10.1101/2020.11.15.378125) -->
          <p>Beltran H., Romanel A., â€¦ , <strong>Dann E.</strong>, â€¦ , Benelli M., and Demichelis F. (2020) Circulating Tumor DNA Profile Recognizes Transformation to Castration-Resistant Neuroendocrine Prostate Cancer, <em>Journal of Clinical Investigation</em> | <a href="https://www.jci.org/articles/view/131041">Full text</a></p>
          </div>
        </div>
      </div>
    </div>

    <div class="container d-block d-lg-none d-xl-none">
      <div class="d-flex flex-row justify-content-center">
        <div class="d-flex flex-col">
          <div class="mx-auto">
            <div class="p-3 mt-3 text-center">
              <img src="" style="height:15rem" class="rounded">
            </div>
            <div class="p-2 text-center">
              <h1>Publications ðŸ“œ</h1>
            </div>
            <hr style="width:60%;"/>

            <div class="row justify-content-center align-items-center">
              <div class="col-8 p-1 text-center">
                <div class="list-group">
                                  </div>
              </div>
            </div>
            <div class="mt-4 mb-5 mx-2">
              <div id="key-publications" class="section level4">
              <h4>Key publications</h4>
              <p><strong>Dann E.</strong>, Teichmann S.A., Marioni J.C. (2022) Precise identification of cell states altered in disease with healthy single-cell references, <em>biorXiv (accepted at Nature Genetics)</em> | <a href="https://www.biorxiv.org/content/10.1101/2022.11.10.515939v1">Full text</a> | <a href="https://github.com/MarioniLab/oor_design_reproducibility">code</a> | <a href="https://github.com/MarioniLab/oor_benchmark">python package</a></p>
              <p>Suo C.*, <strong>Dann E.</strong>*, â€¦ , Haniffa M., Teichmann S.A. (2022) Mapping the developing human immune system across organs. <em>Science</em> | <a href="https://www.science.org/doi/10.1126/science.abo0510#con2">Full text</a> | <a href="https://github.com/Teichlab/Pan_fetal_immune">code</a> | <a href="https://developmental.cellatlas.io/fetal-immune">data</a><br />
              <sup>*Equal Contribution</sup></p>
              <p><strong>Dann E.</strong>, Henderson N.C., Teichmann S.A., Morgan M.D., Marioni J.C. (2021) Differential abundance testing on single-cell data using k-nearest neighbor graphs, <em>Nature Biotechnology</em> | <a href="https://www.nature.com/articles/s41587-021-01033-z">Full text</a> | <a href="https://github.com/MarioniLab/miloR">R package</a> | <a href="https://github.com/emdann/milopy">python package</a></p>
              </div>
              <div id="other" class="section level4">
              <h4>Other</h4>
              <p>Schuster V., <strong>Dann E.</strong>, Krogh A., Teichmann S.A. (2023) multiDGD: A versatile deep generative model for multi-omics data. <em>biorXiv</em> | <a href="https://www.biorxiv.org/content/10.1101/2023.08.23.554420v1.full">Full text</a> | <a href="https://github.com/Center-for-Health-Data-Science/multiDGD">python package</a></p>
              <p>Suo C., Polanski K., <strong>Dann E.</strong>, â€¦ , Tuong Z.K., Clatworthy M., Teichmann S.A. (2022) Dandelion uses the single-cell adaptive immune receptor repertoire to explore lymphocyte developmental origins, <em>Nature Biotechnology</em> | <a href="https://www.nature.com/articles/s41587-023-01734-7">Full text</a> | <a href="https://github.com/zktuong/dandelion">python package</a></p>
              <p>Heumos L., â€¦ <strong>Single-cell Best Practices Consortium</strong>, â€¦ , Theis F.J. (2023) Best practices for single-cell analysis across modalities. <em>Nature Review Genetics</em> | <a href="https://www.nature.com/articles/s41576-023-00586-w">full text</a></p>
              <p>Missarova A., <strong>Dann E.</strong>, â€¦ , J.C. Marioni (2023) Sensitive cluster-free differential expression testing. <em>biorXiv</em> | <a href="https://www.biorxiv.org/content/10.1101/2023.03.08.531744v1">Full text</a> | <a href="https://github.com/MarioniLab/miloDE">R package</a></p>
              <p>Sumanaweera D., Suo C., Muraro D., <strong>Dann E.</strong>, â€¦ , Teichmann S.A. (2023) Gene-level alignment of single cell trajectories informs the progression of in vitro T cell differentiation. <em>biorXiv</em> | <a href="https://www.biorxiv.org/content/10.1101/2023.03.08.531713v1">Full text</a> | <a href="https://github.com/Teichlab/Genes2Genes">python package</a></p>
              <p>Barnes J. L., He P., â€¦ , <strong>Dann E.</strong>, â€¦ , Teichmann S.A., Meyer K. B., Nikolic M. Z. (2022) Early human lung immune cell development and its role in epithelial cell fate. <em>biorXiv</em> | <a href="https://www.biorxiv.org/content/biorxiv/early/2022/12/15/2022.12.13.519713.full.pdf">Full text</a></p>
              <p>He P., Lim K., Sun D., â€¦ , <strong>Dann E.</strong>, â€¦ , Meyer K.B., Rawlins E.L. (2022) A human fetal lung cell atlas uncovers proximal-distal gradients of differentiation and key regulators of epithelial fates, <em>Cell</em> | <a href="https://www.cell.com/cell/fulltext/S0092-8674(22)01415-5?_returnURL=https%3A%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS0092867422014155%3Fshowall%3Dtrue">Full text</a></p>
              <p>Lalchand V.*, Ravuri A.*, <strong>Dann E.</strong>*, â€¦ , Teichmann S.A., Lawrence N.D. (2022) Modelling Technical and Biological Effects in scRNA-seq data with Scalable GPLVMs. <em>Proceedings of the 17th Machine Learning in Computational Biology meeting</em> | <a href="https://proceedings.mlr.press/v200/lalchand22a">Full text</a><br />
              <sup>*Equal Contribution</sup></p>
              <p>Botting R.A., Goh I., â€¦ , <strong>Dann E.</strong>, â€¦ , Teichmann S.A. , Haniffa M. (2022) Multi-organ functions of yolk sac during human early development. <em>biorXiv</em> | <a href="https://www.biorxiv.org/content/10.1101/2022.08.03.502475v1.full.pdf">Full text</a></p>
              <p>Kleshchevnikov V., Shmatko A., <strong>Dann E.</strong>, â€¦, Stegle O., Bayraktar O.A. (2022) Cell2location maps fine-grained cell types in spatial transcriptomics. <em>Nature Biotechnology</em> | <a href="https://doi.org/10.1038/s41587-021-01139-4">Full text</a> | <a href="https://cell2location.readthedocs.io/">python package</a></p>
              <p>Jardine L., Webb S., â€¦ , <strong>Dann E.</strong>, â€¦ , GÃ¶ttgens B., Roberts I., Teichmann S. and Haniffa M. (2021) Blood and immune development in human fetal bone marrow and Down syndrome, <em>Nature</em> | <a href="https://doi.org/10.1038/s41586-021-03929-x">Full text</a></p>
              <p>Elmentaite R., Kumasaka N., Roberts K., Fleming A., <strong>Dann E.</strong>, â€¦ , James K.R., Teichmann S.A. (2021), Cells of the human intestinal tract mapped across space and time, <em>Nature</em> | <a href="https://www.nature.com/articles/s41586-021-03852-1">Full text</a></p>
              <p>Stephenson E., Reynolds G., Botting R.A., Calero-Nieto F.J., Morgan M.D., Tuong Z.K., Bach K., Sungnak W., â€¦ , <strong>Dann E.</strong>, â€¦ , Duncan C.J.A, Smith K., Teichmann S.A., Clatworthy M.R., Marioni J.C., Gottgens B., Haniffa M. (2020) Single-cell multi-omics analysis of the immune response in COVID-19, <em>Nature Medicine</em> | <a href="https://www.nature.com/articles/s41591-021-01329-2">Full text</a></p>
              <!-- Kleshchevnikov V., Shmatko A., **Dann E.**, ..., Stegle O.,  Bayraktar O.A. (2020) Comprehensive mapping of tissue cell architecture via integrated single cell and spatial transcriptomics, bioRxiv, [doi: https://doi.org/10.1101/2020.11.15.378125](https://doi.org/10.1101/2020.11.15.378125) -->
              <p>Beltran H., Romanel A., â€¦ , <strong>Dann E.</strong>, â€¦ , Benelli M., and Demichelis F. (2020) Circulating Tumor DNA Profile Recognizes Transformation to Castration-Resistant Neuroendocrine Prostate Cancer, <em>Journal of Clinical Investigation</em> | <a href="https://www.jci.org/articles/view/131041">Full text</a></p>
              </div>
            </div>
        </div>
      </div>
    </div>

    <!-- Optional JavaScript -->
    <!-- jQuery first, then Popper.js, then Bootstrap JS -->
    <script src="https://code.jquery.com/jquery-3.5.1.slim.min.js" integrity="sha384-DfXdz2htPH0lsSSs5nCTpuj/zy4C+OGpamoFVy38MVBnE+IbbVYUew+OrCXaRkfj" crossorigin="anonymous"></script>
    <script src="https://cdn.jsdelivr.net/npm/popper.js@1.16.0/dist/umd/popper.min.js" integrity="sha384-Q6E9RHvbIyZFJoft+2mJbHaEWldlvI9IOYy5n3zV9zzTtmI3UksdQRVvoxMfooAo" crossorigin="anonymous"></script>
    <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.5.0/js/bootstrap.min.js" integrity="sha384-OgVRvuATP1z7JjHLkuOU7Xw704+h835Lr+6QL9UvYjZE3Ipu6Tp75j7Bh/kR0JKI" crossorigin="anonymous"></script>

    
  </body>
</html>
